core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -482,6 +482,10 @@ public void onApplicationEvent(ApplicationEvent event) {
 
 		if (refresh) {
 			binderProducingContext.refresh();
+			if (this.context != null) {
```

Comment:
I noticed this was added just now (therefore not actually needed by the other binders). Is this just so that we can get access to the context in the tests? If so, could we not just verify the properties are set on the actual consumer/producer like the Kafka Binder ITs do for this?

### Example
The Producer key serializer gets set via default property here
https://github.com/spring-cloud/spring-cloud-stream/blob/e06d4eedfd70302f0ba7241adb3e3e3d8d8b00d5/binders/kafka-binder/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/integration/KafkaBinderExtendedPropertiesTest.java#L63-L64

and then is verified that it is on the binding properties https://github.com/spring-cloud/spring-cloud-stream/blob/e06d4eedfd70302f0ba7241adb3e3e3d8d8b00d5/binders/kafka-binder/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/integration/KafkaBinderExtendedPropertiesTest.java#L92-L97

Reply:
@onobc We can do that, but I like to avoid those casting and the usage of DBF to gain access to the binder, etc. I had a discussion with @olegz on this the other day, and we decided to make this change at the core level to add the binder context to the parent context as a separate isolated context for anyone to get this type of read-only access to the binder context. Post 4.1 release, we were even thinking of removing this parent-child hierarchy altogether for scenarios that don't need that type of relationship (multi-binder is the only thing that requires it). If you are okay with that, l say, let's go with this approach. Please let me know.

Reply:
Makes sense. I do also ❤️ the idea of scrapping the parent/child when not needed as it add complexity.